bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/target/env/Asterism.scala

```diff
@@ -1,31 +1,66 @@
 package edu.gemini.spModel.target.env
 
-import edu.gemini.spModel.core.Coordinates
+import edu.gemini.spModel.core.{Coordinates, NonSiderealTarget, ProperMotion, SiderealTarget, Target}
 import edu.gemini.spModel.target.SPTarget
-import edu.gemini.shared.util.immutable.{Option => GOption}
+import edu.gemini.shared.util.immutable.{ImList, Option => GOption}
 import edu.gemini.shared.util.immutable.ScalaConverters._
 import edu.gemini.skycalc.{Coordinates => SCoordinates}
-
-
 import java.time.Instant
 
-import scalaz.NonEmptyList
+import scalaz._
+import Scalaz._
 
 /** Collection of stars that make up the science target(s) for an observation,
   * along with any configuration details unique to the instrument in use.
   */
 trait Asterism {
 
-  /** All targets that comprise the asterism.  There must be at least one target.
+  // invariant: allSpTargets.length == targets.fold(_ => 1, _ => 2)
+
+  /** All SPTargets that comprise the asterism. */
+  def allSpTargets: NonEmptyList[SPTarget]
+
+  /** All SPTargets that comprise the asterism, as a Gemini ImList. */
+  def allSpTargetsJava: ImList[SPTarget] =
+    allSpTargets.list.toList.asImList
+
+  /** All Targets that comprise the asterism. */
+  def allTargets: NonEmptyList[Target] =
+    targets.fold(NonEmptyList(_), p => NonEmptyList(p._1, p._2))
+
+  /** An asterism is a single generic target, or a pair of sidereal targets. These are currently the
+    * only possibilities.
+    * TODO: should we Church encode this instead?
     */
-  def targets: NonEmptyList[SPTarget]
+  def targets: Target \/ (SiderealTarget, SiderealTarget)
```

Comment:
I suppose it makes sense to worry about the three target case, or whatever else is coming, when it happens.  This will make it easier to make all the other changes that need to be made to support switching out the base position with an asterism without.  It seems to me at some point though we'll have to punt and match on the asterism.